src/main.rs

```diff
@@ -0,0 +1,223 @@
+use libc::{uid_t, gid_t};
+use std::io::prelude::*;
+use std::ffi::CString;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+use std::process::exit;
+
+struct BindMount {
+	dest: PathBuf,
+	src: PathBuf
+}
+
+struct Config {
+	rootfs: PathBuf,
+	uid: uid_t,
+	gid: gid_t,
+	args: Vec<String>,
+	bind_mounts: Vec<BindMount>,
+}
+
+// TODO: This function does not really perform error checking;
+//       for now, we assume that xbstrap passes sane values.
+fn make_config_from_cli() -> Config {
+	let matches = clap::App::new("cbuildrt")
+		.arg(clap::Arg::with_name("cbuild-json")
+			.help("cbuild.json file")
+			.required(true)
+		)
+		.get_matches();
+
+	let cfg_f = File::open(matches.value_of("cbuild-json").unwrap())
+		.expect("unable to open cbuild.json");
+	let cfg : serde_json::Value = serde_json::from_reader(cfg_f)
+		.expect("failed to parse cbuild.json");
+
+	return Config{
+		rootfs: PathBuf::from(cfg["rootfs"].as_str().unwrap()),
+		uid: cfg["user"]["uid"].as_u64().unwrap() as uid_t,
+		gid: cfg["user"]["gid"].as_u64().unwrap() as gid_t,
+		args: cfg["process"]["args"].as_array().unwrap().iter()
+			.map(|a| a.as_str().unwrap().to_owned())
+			.collect::<Vec<_>>(),
+		bind_mounts: cfg["bindMounts"].as_array().unwrap().iter()
+			.map(|bm| {
+				//println!("{}", bm["destination"]);
+			BindMount{
+				dest: PathBuf::from(bm["destination"].as_str().unwrap()),
+				src: PathBuf::from(bm["source"].as_str().unwrap())
+			}})
+			.collect::<Vec<_>>()
+	};
+}
+
+// Concatenates lhs and rhs as-if the rhs was a relative path.
+fn concat_absolute<L: AsRef<Path>, R: AsRef<Path>> (lhs: L, rhs: R) -> PathBuf {
+	let mut buf = lhs.as_ref().to_path_buf();
```

Comment:
I think this function can be `lhs.concat(rhs.strip_prefix("/").unwrap())` or something